likecoin/admin/views/matters.php

```diff
@@ -94,13 +94,81 @@ function likecoin_append_footer_link_element( $dom_document ) {
 	$dom_document->documentElement->appendChild( $p );
 }
 
+/**
+ * Parse and modify post HTML to replace Matters asset url and div/class standard
+ *
+ * @param string|  $post_id raw post HTML content.
+ * @param WP_Post| $post Post object.
+ */
+function likecoin_upload_url_image_to_matters( $post_id, $post ) {
+	if ( ! $post ) {
+		return;
+	}
+	$content = $post->post_content;
+	if ( ! $content ) {
+		return $content;
+	}
+	$dom_document          = new DOMDocument();
+	$libxml_previous_state = libxml_use_internal_errors( true );
+	$dom_content           = $dom_document->loadHTML( '<template>' . mb_convert_encoding( $content, 'HTML-ENTITIES' ) . '</template>', LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+	libxml_clear_errors();
+	libxml_use_internal_errors( $libxml_previous_state );
+	if ( false === $dom_content ) {
+		return $content;
+	}
+	$images             = $dom_document->getElementsByTagName( 'img' );
+	$current_image_urls = array();
```

Comment:
Since we want to access by url to check for existence later, a Set would be more suitable. But php does not have a native set, I think using a key-value array like `$image_infos` is better here.